src/lib/device/chunk.cpp

```diff
@@ -0,0 +1,71 @@
+/*********************************************************************************
+ * Modifications Copyright 2017-2019 eBay Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ *********************************************************************************/
+#include "device/chunk.h"
+#include "device/device.h"
+#include "device/physical_dev.hpp"
+#include "common/homestore_utils.hpp"
+
+namespace homestore {
+Chunk::Chunk(PhysicalDev* pdev, const chunk_info& cinfo, uint32_t chunk_slot) :
+        m_chunk_info{cinfo}, m_pdev{pdev}, m_chunk_slot{chunk_slot}, m_stream_id{pdev->chunk_to_stream_id(cinfo)} {}
+
+void Chunk::cp_flush() {}
+
+std::string Chunk::to_string() const {
+    return fmt::format("chunk_id={}, vdev_id={}, start_offset={}, size={}, end_of_chunk={}, slot_num_in_pdev={} "
+                       "pdev_ordinal={} vdev_ordinal={} stream_id={}",
+                       chunk_id(), vdev_id(), start_offset(), in_bytes(size()), end_of_chunk(), slot_number(),
+                       pdev_ordinal(), vdev_ordinal(), stream_id());
+}
+
+void Chunk::set_user_private(const sisl::blob& data) {
+    std::unique_lock lg{m_mgmt_mutex};
+    m_chunk_info.set_user_private(data);
+    m_chunk_info.compute_checksum();
+
+    auto buf = hs_utils::iobuf_alloc(chunk_info::size, sisl::buftag::superblk, physical_dev()->align_size());
+    auto cinfo = new (buf) chunk_info();
+    *cinfo = m_chunk_info;
+    physical_dev_mutable()->write_super_block(buf, chunk_info::size,
+                                              physical_dev()->chunk_info_offset_nth(slot_number()));
+    cinfo->~chunk_info();
```

Comment:
m_chunk_info is not aligned buffer and we need aligned iobuf memory to write.

Reply:
where to get the align size? set it manually or get from system command

Reply:
aligned size is reported as part of device attributes from iomanager. It could vary on physical device type. As of now 512 is good enough, that is what iomanager reports.